pkg/config/codegen/parse.go

```diff
@@ -0,0 +1,251 @@
+// and add respective Viper flag 
+func AddFlags(c *cobra.Command, v *viper.Viper) {
+	{{- range . }}
+	c.Flags().{{ .FlagDesc.CobraType }}("{{ .FlagDesc.Name }}", {{ defaultVal .FlagDesc.CobraType }}, "{{ .FlagDesc.Description }}")
+	if err := v.BindPFlag("{{ .FlagDesc.JSONPath }}", c.Flags().Lookup("{{ .FlagDesc.Name }}")); err != nil {
+		panic(err)
+	}
+
+	{{- end }}
+}
+`
+)
+
+var (
+	fileTemplate = template.Must(
+		template.New("gen").
+			Funcs(sprig.TxtFuncMap()).
+			Funcs(template.FuncMap{"defaultVal": mustGetDefaultValueRepr}).
+			Parse(fileTemplateStr),
+	)
+)
+
+// These types are used in template to add appropriate Cobra flags
+var (
+	stringType  = newFlagType("String", `""`)
+	intType     = newFlagType("Int", "0")
+	boolType    = newFlagType("Bool", "false")
+	float32Type = newFlagType("Float32", "0")
+	float64Type = newFlagType("Float64", "0")
+
+	stringSliceType = newFlagType("StringSlice", "nil")
+
+	defaultValueReprsByTypeName = make(map[string]string)
+)
+
+func mustGetDefaultValueRepr(typeName string) string {
+	valueRepr := defaultValueReprsByTypeName[typeName]
+	if valueRepr == "" {
+		panic(fmt.Sprintf("couldn't represent default value for type %s", typeName))
+	}
+	return valueRepr
+}
+
+func newFlagType(typeName, defaultValueRepr string) string {
+	defaultValueReprsByTypeName[typeName] = defaultValueRepr
+	return typeName
+}
+
+type flagDesc struct {
+	Name        string
+	JSONPath    string
+	Description string
+	CobraType   string
+}
+
+type templateElem struct {
+	FlagDesc flagDesc
+}
+
+func getJSONKey(member types.Member) (string, error) {
+	jsonTag := reflect.StructTag(member.Tags).Get("json")
+	if jsonTag == "" {
+		return "", errors.Errorf("member %v does not specify a json tag", member)
+	}
+	return strings.Split(jsonTag, ",")[0], nil
+}
+
+func getDescription(member types.Member) string {
+	firstCommentLineWithMetadata := len(member.CommentLines)
+	for i, commentLine := range member.CommentLines {
+		if strings.HasPrefix(commentLine, metadataMarker) {
+			firstCommentLineWithMetadata = i
+			break
+		}
+	}
+	return strings.Join(member.CommentLines[:firstCommentLineWithMetadata], " ")
+}
+
+func getCobraTypeFromMember(memberType *types.Type) string {
+	switch memberType.Kind {
+	case types.Builtin:
+		switch memberType {
+		case types.String:
+			return stringType
+		case types.Int:
+			return intType
+		case types.Float32:
+			return float32Type
+		case types.Float64:
+			return float64Type
+		case types.Bool:
+			return boolType
+		}
+	case types.Slice:
+		if memberType.Elem == types.String {
+			return stringSliceType
+		}
+	}
+	return ""
+}
+
+func constructFlagDescsFromStruct(typeSpec *types.Type, jsonPathSoFar []string) ([]flagDesc, error) {
+	var flagDescs []flagDesc
+	for _, member := range types.FlattenMembers(typeSpec.Members) {
+		extractedTags := types.ExtractCommentTags("+", member.CommentLines)
+		flagNameTags := extractedTags[flagNameTagKey]
+		if len(flagNameTags) > 1 {
```

Comment:
Does `types.ExtractCommentTags` handle comma-separated list? If not, and since we are anticipating 1:1 mapping, maybe we should handle that.

Reply:
No, it doesn't treat comma separated lists specially. If a user wants to specify multiple values, they need to do it in multiple lines. Does this make sense? What do you mean by handling that in this context?

Reply:
Got it.